src/decide.rs

```diff
@@ -0,0 +1,37 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Debug, PartialEq, Eq)]
+pub struct Decide {
+    pub distinct_id: String,
+}
+
+impl Decide {
+    pub fn new<S: Into<String>>(distinct_id: S) -> Self {
```

Comment:
What is the reason for these generics? Since we are not using it in the public interface we don't need to be this careful :)

Reply:
It's a copy-paste from the original repo https://github.com/PostHog/posthog-rs/blob/main/src/lib.rs#L123-L129. But Event can be constructed by a user of the crate. I had the intention to have Decide be used in the same way when I thought group-types would be a part of the request. But then I removed group-types, so I can change the generic argument :)